### DISCUSSION TITLE
Question about uv_get_available_memory API

### DISCUSSION THREAD

**Discussion by theanarkh** [libuv/libuv#4336] [Q&A] [answered]
The value of `uv_get_available_memory` is equal to `limit_in_bytes - usage_in_bytes`, but the value of `usage_in_bytes` includes `RSS` and `Page Cache`. So i think the available memory  should equal to `limit_in_bytes - RSS` instead of `limit_in_bytes - usage_in_bytes` ?

**Comment by bnoordhuis** [comment]
I don't really understand what you're asking. If you're saying libuv is doing something wrong, please motivate why.

**Comment by vtjnash** [comment]
I think it is a reference to https://github.com/libuv/libuv/issues/3897, but I am increasingly uncertain that is better than the (also fairly useless) status quo.

**Reply by theanarkh** [reply]
Thanks !

**Comment by theanarkh** [comment]
I want to know how much memory the process can use before OOM.

**Reply by bnoordhuis** [reply, marked as answer]
On general-purpose multitasking operating systems that's fundamentally unknowable. uv_get_available_memory() is an educated guess, useful for diagnostics but not for policy. Reasons why:

- the system's available resources may have changed by the time the function returns
- your process may not be able or is not allowed to allocate that much memory
- your process can allocate way more due to overcommit or (possibly compressed) swap

If you're trying to write software that's robust in low memory situations, then your best best is to disable overcommit and always handle allocation failures, but even that may not always stave off the OOM killer.

**Reply by theanarkh** [reply]
Thanks for the clarification.

**Reply by theanarkh** [reply]
@bnoordhuis  If there are two processes in the container.
`limit_in_bytes` indicates `memory of process1` + `memory of process2` cannot exceed the `limit_in_bytes limit` ?
`usage_in_bytes` Indicates that `memory of process1` + `memory of process2` is equal to `usage_in_bytes` ?

**Reply by bnoordhuis** [reply]
Yes, I believe that's correct.

**Reply by theanarkh** [reply]
Thanks !
